src/mscorlib/shared/System/String.Searching.cs

```diff
@@ -357,7 +355,10 @@ public int IndexOf(string value, int startIndex, int count, StringComparison com
         //
         public int LastIndexOf(char value)
         {
-            return LastIndexOf(value, this.Length - 1, this.Length);
+            if (Length == 0)
```

Comment:
If we do not have Length == 0 check for IndexOf, we should not have one here either.